src/app/services/taskDetailsApi.ts

```diff
@@ -1,14 +1,48 @@
 import { TASKS_URL } from '@/constants/url';
 import { api } from './api';
 import { taskDetailsDataType } from '@/interfaces/taskDetails.type';
+import { FetchBaseQueryError } from '@reduxjs/toolkit/dist/query';
+
+type dependency = PromiseSettledResult<{
+    title: string | undefined;
+    id: string;
+}>[];
 
 export const taskDetailsApi = api.injectEndpoints({
     endpoints: (build) => ({
         getTaskDetails: build.query<taskDetailsDataType, string>({
             query: (taskId): string => `${TASKS_URL}/${taskId}/details`,
             providesTags: ['Task_Details'],
         }),
+        getTasksDependencyDetails: build.query<dependency, string[]>({
+            queryFn: async (taskIds, _queryApi, _extraOptions, baseQuery) => {
+                try {
+                    const taskDetailsPromise = taskIds.map(async (taskId) => {
+                        const result = await baseQuery(
+                            `${TASKS_URL}/${taskId}/details`
+                        );
+
+                        if (result.error) {
+                            throw { error: result.error, id: taskId };
```

Comment:
you are throwing the error from here but are you handling it at the parent level ?